### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by ruffsl**
```
docker pull rosswg/turtlebot3_demo:dashing
docker pull rosswg/turtlebot3_demo:nightly
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# build underlay package source
+ARG CMAKE_BUILD_TYPE=Release
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+    colcon build \
+      --symlink-install \
+      --cmake-args \
+        -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
+        -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+        -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
```
Comment (mikaelarguedas) [inline Dockerfile:32]: We could use some mixins here (same below):
https://github.com/colcon/colcon-mixin-repository/blob/master/ccache.mixin
https://github.com/colcon/colcon-mixin-repository/blob/master/build-type.mixin

Comment (ruffsl) [reply Dockerfile:32]: Do mixins 'mix well' with environment variables and build args?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+# source overlay workspace from entrypoint
+RUN sed --in-place \
+      's|^source .*|source "$TB3_WS/install/setup.bash"|' \
+      /ros_entrypoint.sh
+
+WORKDIR $TB3_WS
+COPY ./maps ./policies ./
+ENV TURTLEBOT3_MODEL burger
+ENV TB3_POLICY_FILE $TB3_WS/policies/tb3_gazebo_policy.xml
```
Comment (o-olalde) [inline Dockerfile:69]:
```suggestion
ENV TB3_POLICY_FILE $TB3_WS/policies/tb3_gazebo_policy.xml

# Add security tools
RUN pip3 install aztarna && pip3 install ros2-fuzzer
```
Add commands to install the tools inside the container

File: docker/underlay.repos [hunk 3]
```diff
@@ -0,0 +1,10 @@
+repositories:
+  cartographer/cartographer_ros:
+    type: git
+    url: https://github.com/ROBOTIS-GIT/cartographer_ros.git
+    version: dashing
+  ros-planning/navigation2:
+    type: git
+    url: https://github.com/mikaelarguedas/navigation2.git
+    version: mikael/dashing-sync
+    # using nav 0.2.3 release candidate, current 0.2.2 has amcl / map issues preventing this demo from running
```
Comment (o-olalde) [inline docker/underlay.repos:10]:
```suggestion
    # using nav 0.2.3 release candidate, current 0.2.2 has amcl / map issues preventing this demo from running
  ros2/robot_state_publisher:
    type: git
    url: https://github.com/ros2/robot_state_publisher.git
    version: 2.2.3
  ros2/rviz:
    type: git
    url: https://github.com/ros2/rviz.git
    version: 6.1.3
```
Add dependencies to run the cartographer node

**Review by o-olalde — CHANGES_REQUESTED** [review]
I've added our tools in this pull request after testing it. While locally testing it I've found necessary to add some dependencies here included.

**Comment by mikaelarguedas** [comment]
@ruffsl is it possible to update the existing docker file instead of creating a new one and to describe the changes? Ideally we would also update the documentation to match any change being merged (path changes etc)

**Pull request merged**